sapl/base/forms.py

```diff
@@ -1387,38 +1388,107 @@ class Meta:
 
         self.helper = SaplFormHelper()
         self.helper.layout = Layout(
             row1, row2, row3,
-            form_actions(label='Salvar'))
+            form_actions(label='Salvar')
+        )
+
+    def clean(self):
+        cleaned_data = self.cleaned_data
+
+        if not self.is_valid():
+            return cleaned_data
+
+        if cleaned_data['data_criacao'] and cleaned_data['data_extincao'] and cleaned_data['data_criacao'] > \
+                cleaned_data['data_extincao']:
+            raise ValidationError("Certifique-se de que a data de criação seja anterior à data de extinção.")
+
+        if self.instance.pk:
+            partido = Partido.objects.get(pk=self.instance.pk)
+
+            if xor(cleaned_data['sigla'] == partido.sigla, cleaned_data['nome'] == partido.nome):
+                raise ValidationError(_('O Partido deve ter um novo Nome e uma nova Sigla.'))
+
+            cleaned_data.update({'partido': partido})
+
+        return cleaned_data
+
+class PartidoUpdateForm(PartidoForm):
+
+    opcoes = (('nao','Não'),
+              ('sim','Sim')
+            )
+
+    historico = forms.ChoiceField(initial='nao', choices=opcoes)
+
+
+    class Meta:
+        model = Partido
+        exclude = []
+
+
+    def __init__(self, pk=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        row1 = to_row([
+            ('sigla', 6),
+            ('nome', 6),
+            ]
+        )
+        row2 = to_row([
+            ('historico', 2),
+            ('data_criacao', 5),
+            ('data_extincao', 5),
+            ]
+        )
+        row3 = to_row([('observacao', 12)])
+        row4 = to_row([('logo_partido', 12)])
+
+        self.helper = SaplFormHelper()
+        self.helper.layout = Layout(
+            row1, row2, row3, row4,
+            form_actions(label='Salvar')
+        )
 
     def clean(self):
+        cleaned_data = self.cleaned_data
 
-        cleaned_data = super(PartidoForm, self).clean()
+        is_historico = cleaned_data['historico'] == 'sim' 
 
         if not self.is_valid():
             return cleaned_data
-            
-        if cleaned_data['data_criacao'] and cleaned_data['data_extincao']:
-            if cleaned_data['data_criacao'] > cleaned_data['data_extincao']:
-                raise ValidationError("Certifique-se de que a data de criação seja anterior à data de extinção.")
 
-        return cleaned_data
\ No newline at end of file
+        if cleaned_data['data_criacao'] and cleaned_data['data_extincao'] and cleaned_data['data_criacao'] > \
+                cleaned_data['data_extincao']:
+                raise ValidationError("Certifique-se de que a data de inicio seja anterior à data fim.")
+
+        if is_historico: 
+            if cleaned_data['data_criacao'] == None or cleaned_data['data_extincao'] == None:
+                raise ValidationError("Certifique-se de que a data de inicio e fim de historico estão preenchidas")
```

Comment:
Esse if pode ser feito como
`if not cleaned_data['data_criacao']  or not cleaned_data['data_extincao']:`

Reply:
Resolvido.